network/p2p/gossip/metrics.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+const (
+	ioLabel    = "io"
+	sentIO     = "sent"
+	receivedIO = "received"
+
+	typeLabel  = "type"
+	pushType   = "push"
+	pullType   = "pull"
+	unsentType = "unsent"
+	sentType   = "sent"
+
+	// dropped indicate that the gossipable element was not added to the set due to some reason.
+	// for sent message, we'll use notReceive below.
+	droppedLabel = "dropped"
+
+	droppedMalformed = "malformed"
+	DroppedDuplicate = "duplicate"
```

Comment:
Why is this exported?

Reply:
We don't *have to* export this label. Beside the code inside the gossip package, we use it in the coreth implementation along with the `ObserveIncomingGossipable` call. We can definitely define a separate label text ( doesn't have to be `duplicate` ). As a starting point, I think that keeping it the same is good enough. We can always change that later on if we see that it's unused/unjustified/etc.